extension/src/openvic-extension/singletons/MapItemSingleton.cpp

```diff
@@ -0,0 +1,229 @@
+
+
+//billboards, projections, and progress bar
+
+#include "MapItemSingleton.hpp"
+#include <string_view>
+
+#include <godot_cpp/variant/utility_functions.hpp>
+
+#include <openvic-simulation/map/ProvinceInstance.hpp>
+
+#include "openvic-extension/singletons/GameSingleton.hpp"
+#include "openvic-extension/utility/ClassBindings.hpp"
+#include "openvic-extension/utility/Utilities.hpp"
+#include "godot_cpp/variant/packed_int32_array.hpp"
+#include "godot_cpp/variant/packed_vector2_array.hpp"
+#include "godot_cpp/variant/typed_array.hpp"
+#include "godot_cpp/variant/vector2.hpp"
+#include "openvic-simulation/DefinitionManager.hpp"
+#include "openvic-simulation/country/CountryDefinition.hpp"
+#include "openvic-simulation/country/CountryInstance.hpp"
+#include "openvic-simulation/interface/GFXObject.hpp"
+#include "openvic-simulation/map/ProvinceDefinition.hpp"
+#include "openvic-simulation/types/Vector.hpp"
+
+using namespace godot;
+using namespace OpenVic;
+
+void MapItemSingleton::_bind_methods() {
+	OV_BIND_METHOD(MapItemSingleton::get_billboards);
+	OV_BIND_METHOD(MapItemSingleton::get_province_positions);
+	OV_BIND_METHOD(MapItemSingleton::get_province_count);
+	OV_BIND_METHOD(MapItemSingleton::get_capital_count);
+	OV_BIND_METHOD(MapItemSingleton::get_capital_positions);
+	OV_BIND_METHOD(MapItemSingleton::get_crime_icons);
+	OV_BIND_METHOD(MapItemSingleton::get_rgo_icons);
+}
+
+MapItemSingleton* MapItemSingleton::get_singleton() {
+	return singleton;
+}
+
+MapItemSingleton::MapItemSingleton() {
+	ERR_FAIL_COND(singleton != nullptr);
+	singleton = this;
+}
+
+MapItemSingleton::~MapItemSingleton() {
+	ERR_FAIL_COND(singleton != this);
+	singleton = nullptr;
+}
+
+// Get the billboard object from the loaded objects
+
+GFX::Billboard const* MapItemSingleton::get_billboard(std::string_view name, bool error_on_fail) const {
+	GameSingleton const* game_singleton = GameSingleton::get_singleton();
+	ERR_FAIL_NULL_V(game_singleton, nullptr);
+	DefinitionManager const* definition_manager = &game_singleton->get_definition_manager();
+	ERR_FAIL_NULL_V(definition_manager, nullptr);
+
+	GFX::Billboard const* billboard =
+		definition_manager->get_ui_manager().get_cast_object_by_identifier<GFX::Billboard>(name);
+
+	if (error_on_fail) {
+		//ERR_FAIL_NULL_V_MSG(billboard, nullptr, vformat("Failed to find billboard \"%s\"", Utilities::std_to_godot_string(name)));
+		ERR_FAIL_NULL_V_MSG(billboard, nullptr, "Failed to find billboard inner");
```

Comment:
```suggestion
		ERR_FAIL_NULL_V_MSG(billboard, nullptr, vformat("Failed to find billboard \"%s\"", Utilities::std_to_godot_string(name)));
```
Might as well use the error message including the name of the billboard that wasn't found, and remove the other error message.